cmd/agent/config.go

```diff
@@ -44,17 +45,31 @@ func LoadConfig() (*agent.Config, error) {
 		return nil, errors.New("ssh-address is required, cannot start agent")
 	}
 
+	collectorConfig := &collector.Config{
+		CollectorHost: viper.GetString("collector-host"),
+		CollectorPort: viper.GetInt("collector-port"),
+		EnablemTLS:    enablemTLS,
+		Cert:          cert,
+		Key:           key,
+		CA:            ca,
+	}
+
+	discoveryPeriodsConfig := &discovery.DiscoveriesPeriodConfig{
+		Cluster:      time.Duration(viper.GetInt("cluster-discovery-period")) * time.Second,
```

Comment:
Here maybe we could use something like `GetDuration`: https://github.com/spf13/pflag/blob/master/duration.go#L47
More examples: https://golang.hotexamples.com/examples/github.com.ogier.pflag/-/DurationVar/golang-durationvar-function-examples.html
Maybe with this, we can get rid of further validation. I didn't check, but doing the validation in the way is done now looks a bit ugly hehe